test/Datadog.Trace.ClrProfiler.Native.Tests/metadata_builder_tests.cpp

```diff
@@ -0,0 +1,90 @@
+#include "pch.h"
+
+#include "../../src/Datadog.Trace.ClrProfiler.Native/iterators.h"
+#include "../../src/Datadog.Trace.ClrProfiler.Native/metadata_builder.h"
+
+class MetadataBuilderTest : public ::testing::Test {
+ protected:
+  ModuleMetadata* moduleMetadata_;
+  MetadataBuilder* metadataBuilder_;
+
+  void SetUp() override {
+    ICLRMetaHost* metahost = NULL;
+    HRESULT hr;
+    hr = CLRCreateInstance(CLSID_CLRMetaHost, IID_ICLRMetaHost,
+                           (VOID**)&metahost);
+    ASSERT_FALSE(FAILED(hr));
```

Comment:
Not a big deal, but there's a `SUCCEEDED(hr)` macro that could make this code a little more readable than these double negatives.